src/app/clusters/operational-credentials-server/operational-credentials-server.cpp

```diff
@@ -0,0 +1,271 @@
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/****************************************************************************
+ * @file
+ * @brief Implementation for the Operational Credentials Cluster
+ ***************************************************************************/
+
+#include <app/Command.h>
+#include <app/server/Server.h>
+#include <app/util/af.h>
+#include <app/util/attribute-storage.h>
+#include <lib/core/PeerId.h>
+#include <platform/CHIPDeviceLayer.h>
+#include <support/CodeUtils.h>
+#include <support/logging/CHIPLogging.h>
+#include <transport/AdminPairingTable.h>
+
+#include "gen/af-structs.h"
+#include "gen/attribute-id.h"
+#include "gen/attribute-type.h"
+#include "gen/cluster-id.h"
+#include "gen/command-id.h"
+
+using namespace chip;
+using namespace ::chip::DeviceLayer;
+using namespace ::chip::Transport;
+
+/*
+ * Temporary flow for fabric management until addOptCert + fabric index are implemented:
+ * 1) When Commissioner pairs with CHIP device, store device nodeId in Admin Pairing table as NodeId
+ *    and store commissioner nodeId in Admin Pairing table as FabricId (This is temporary until AddOptCert is implemented and
+ * Fabrics are implemented correctely) 2) When pairing is complete, commissioner calls SetFabric to set the vendorId on the newly
+ * created fabric. The corresponding fabric is found by looking in admin pairing table and finding a fabric that has the matching
+ * commissioner node ID as fabricId + device nodeId as nodeId and an uninitialized vendorId. 3) RemoveFabric uses the passed in
+ * fabricId, nodeId, vendorID to find matching entry and remove it from admin pairing table. Once fabricIndex is implemented, it
+ * should use that instead.
+ */
+
+EmberAfStatus writeFabricAttribute(uint8_t * buffer, int32_t index = -1)
+{
+    EmberAfAttributeSearchRecord record;
+    record.endpoint         = 0;
+    record.clusterId        = ZCL_OPERATIONAL_CREDENTIALS_CLUSTER_ID;
+    record.clusterMask      = CLUSTER_MASK_SERVER;
+    record.manufacturerCode = EMBER_AF_NULL_MANUFACTURER_CODE;
+    record.attributeId      = ZCL_FABRICS_ATTRIBUTE_ID;
+
+    // When reading or writing a List attribute the 'index' value could have 3 types of values:
+    //  -1: Read/Write the whole list content, including the number of elements in the list
+    //   0: Read/Write the number of elements in the list, represented as a uint16_t
+    //   n: Read/Write the nth element of the list
+    //
+    // Since the first 2 bytes of the attribute are used to store the number of elements, elements indexing starts
+    // at 1. In order to hide this to the rest of the code of this file, the element index is incremented by 1 here.
+    // This also allows calling writeAttribute() with no index arg to mean "write the length".
+
+    return emAfReadOrWriteAttribute(&record, NULL, buffer, 0, true, index + 1);
```

Comment:
could you add as comments what NULL, 0 and true are?

Reply:
done